src/TasksApi.ts

```diff
@@ -0,0 +1,34 @@
+import type { App } from 'obsidian';
+import type { Task } from './Task';
+import { TaskModal } from './TaskModal';
+import { taskFromLine } from './Commands/CreateOrEditTaskParser';
+
+export class TasksApi {
+    public static GetApi(app: App) {
+        return {
+            showUi: (): Promise<string> => {
+                return this.showUi(app);
+            },
+        };
+    }
+
+    public static showUi(app: App): Promise<string> {
```

Comment:
(See the main conversation for discussion about a possible name for this.)

I think a JSDocs block on how to call this and what it returns would be really helpful.

I presume that its behaviour will be heavily dependent on the user's own Tasks plugin settings, for example.

Reply:
Good point. I could also declare an interface for the API and document the interface. What do you think?

Reply:
> Good point. I could also declare an interface for the API and document the interface. What do you think?

I am not entirely sure that I understand your question.

And I am pretty new to TS APIs, so guided by what you recommend.

But people do seem to publish blah.d.ts files, is that what you mean? It seems like a convention, and I presume something that is therefore a Good Thing (TM).

Reply:
Currently the return type of `GetApi` defines the interface for the API. We could change this so that `GetApi` returns an explicit interface and document that interface. The interface could easily be added to the docs as code block (would be short).

It would also help with versioning - we only needed to version the interfaces and could have a concrete class that possibly implements multiple interfaces (API versions).

This has nothing to do with generating type definitions (.d.ts files).

Reply:
> Currently the return type of `GetApi` defines the interface for the API. We could change this so that `GetApi` returns an explicit interface and document that interface.

I don't understand that, after several readings, but will wait to see what you settle for in code, as I imagine it be clearer (to me) then.

> The interface could easily be added to the docs as code block (would be short).

Just to say that it is super easy to embed parts of files, and whole files, in the documentation.

<https://publish.obsidian.md/tasks-contributing/Documentation/Embedding+code+and+data+in+documentation>

So, for example, if the demo vault (in `resources/`) was set up with file(s) to demonstrate this in action, they could be directly embedded in the docs.

The great this about this is that it ensures that the docs stay uptodate.

> It would also help with versioning - we only needed to version the interfaces and could have a concrete class that possibly implements multiple interfaces (API versions).

I feel that the main thing is that it is clear to people reading the code and the user guide what the inputs are, and the return type(s) are...

> This has nothing to do with generating type definitions (.d.ts files).

OK.